src/Transport/Seam/MessagePump.cs

```diff
@@ -17,28 +19,19 @@ class MessagePump : IPushMessages, IDisposable
         RepeatedFailuresOverTimeCircuitBreaker circuitBreaker;
         ILog logger = LogManager.GetLogger(typeof(MessagePump));
         string inputQueue;
+        SatelliteTransportAddressCollection satelliteTransportAddresses;
 
         public MessagePump(ITopologySectionManager topologySectionManager, ITransportPartsContainer container)
         {
             this.topologySectionManager = topologySectionManager;
             this.container = container;
+            satelliteTransportAddresses = container.Resolve<ReadOnlySettings>().Get<SatelliteTransportAddressCollection>();
```

Comment:
wouldn't it be better to inject ReadOnlySettings instead of querying the container here?